main/src/library/LazyList.flix

```diff
@@ -14,57 +14,47 @@
  * limitations under the License.
  */
 
+pub enum LazyList[a] {
+    case ENil,
+    case ECons(a, LazyList[a]),
+    case LCons(a, Lazy[LazyList[a]]),
+    case LList(Lazy[LazyList[a]])
+}
```

Comment:
I'm a little unclear on exactly when each of these should be used. It feels like there is some redundancy here, but an explanation/documentation could convince me otherwise :)

Reply:
Especially the `LList` case. It seems like an extra wrapper that's not needed.

Reply:
Could you put the `lazy` on the outside?

```
pub type alias LazyList[a] = Lazy[Node[a]]

pub enum Node[a] {
    case ENil
    case ECons(a, LazyList[a])
}
```

Reply:
The idea, which is actually hard to articulate, is that we need to support a list that can be both eager and lazy. Moreover, its first element should not always be evaluated. That lead me to the above enum. I was also trying to minimize overhead of the representation of an evaluated list. 

I think Matt's suggestion would work, only downside is that a fully evaluated list would contain an extra layer of indirection through Lazy values. 

Anyway, this is something we should discuss at a meeting!